deliver/lib/deliver/upload_metadata.rb

```diff
@@ -675,9 +675,23 @@ def set_app_rating(version, options)
         has_mapped_values = true
         UI.deprecated("Age rating '#{k}' from iTunesConnect has been deprecated. Please replace with '#{v}'")
       end
-      UI.deprecated("You can find more info at https://docs.fastlane.tools/actions/deliver/#reference") if has_mapped_values
 
-      age_rating_declaration = version.fetch_age_rating_declaration
+      # Handle App Store Connect deprecation/migrations of keys/values if possible
+      attributes, deprecation_messages, errors = Spaceship::ConnectAPI::AgeRatingDeclaration.map_deprecations_if_possible!(attributes)
+      deprecation_messages.each do |message|
+        UI.deprecated(message)
+      end
+
+      unless errors.empty?
+        errors.each do |error|
+          UI.error(error)
+          UI.user_error!("There are Age Rating deprecation errors that cannot be solved automatically... Please apply any fixes and try again")
+        end
+      end
```

Comment:
```suggestion
      unless errors.empty?
        errors.each do |error|
          UI.error(error)
        end
        UI.user_error!("There are Age Rating deprecation errors that cannot be solved automatically... Please apply any fixes and try again")
      end
```

I suppose you wanted to let it crash after iterating through all of the possible error messages?

Reply:
@ainame We don't have access to `Fastlane::UI` inside of `spaceship` otherwise I would have done this 😔 I was either going to:
1) raise an error from the method which is why I had a `!`
2) pass in a block that would return the errors so the caller could call `UI`
3) Or just return the errors like I did here

I went with return the errors and the deprecation messages to be displayed in the calling method that has access to `UI`

Reply:
Oh whoops, this was totally my intention 😛